substrate/bin/utils/chain-spec-builder/bin/main.rs

```diff
@@ -46,18 +55,19 @@ fn inner_main() -> Result<(), String> {
 
 	match builder.command {
 		ChainSpecBuilderCmd::Create(cmd) => {
-			let chain_spec_json = generate_chain_spec_for_runtime(&cmd)?;
+			let is_para = cmd.relay_chain.clone().zip(cmd.para_id.clone());
+			let chain_spec_json = match is_para {
+				Some(_) => generate_chain_spec_for_para_runtime(&cmd)?,
```

Comment:
A little rust nit: 

You assert that `cmd.relay_chain` and `cmd.para_id` are both `Some(_)` in this function, and then the `.expect` is in the `generate_chain_spec_for_para_runtime`. A nicer way would be to pass the unwrapped values inside the `Some(_)` to this function to make it more read-able.

Reply:
I reworked this piece:
https://github.com/paritytech/polkadot-sdk/blob/ce556f19e37c358a12204d51d58a4251633d7c24/substrate/bin/utils/chain-spec-builder/src/lib.rs#L480-L489

I also got rid of newly introduced `Extensions` for parachains. IMO merging jsons is simpler, we also have the same chain spec generation flow for solo and para.